src/api/covariates/coral_atlas.py

```diff
@@ -0,0 +1,94 @@
+import datetime
+from concurrent.futures import ThreadPoolExecutor
+
+import requests
+from typing import List, Tuple
+
+
+class CovariateRequestError(Exception):
+    pass
+
+
+class CoralAtlasCovariate:
+    api_url = "https://allencoralatlas.org"
+    radius = 0.025  # in km
+    num_threads = 3
+    BENTHIC_CLASS_TYPE = "benthic"
+    GEOMORPHIC_CLASS_TYPE = "geomorphic"
+
+    def _sqkm_to_sqm(self, area: float) -> float:
+        return area * 1000000
+
+    def _parse_classes(self, classes: List[dict]) -> List[dict]:
+        classes = classes or []
+        _classes = []
+        for _class in classes:
+            cover_m2 = self._sqkm_to_sqm(_class["cover_sqkm"])
+            _classes.append(dict(name=_class["class_name"], area=cover_m2))
+
+        return sorted(_classes, key=lambda x: (x["area"], x["name"]), reverse=True)
+
+    def _fetch(
+        self, x: float, y: float, radius: float, request_datetime: datetime
+    ) -> List[dict]:
+        url = f"{self.api_url}/mapping/querypoint/{y}/{x}?radius={radius}"
+        resp = requests.get(url)
+        status_code = resp.status_code
+        if status_code != 200:
+            raise CovariateRequestError(resp.text)
+
+        data = (resp.json() or {}).get("data")
+        stats = (data or {}).get("stats")
+        map_assets = (stats or {}).get("map_assets") or []
+        output = {"aca_benthic": [], "aca_geomorphic": []}
+        for map_asset in map_assets:
+            class_type = (map_asset.get("class_type") or "").lower()
+            if not class_type or class_type not in (
+                self.BENTHIC_CLASS_TYPE,
+                self.GEOMORPHIC_CLASS_TYPE,
+            ):
+                continue
+            classes = self._parse_classes(map_asset.get("classes"))
+            output[f"aca_{class_type}"].extend(classes)
+
+        return dict(
```

Comment:
Shouldn't this be a list of dicts?

Reply:
this only deals with one location and all the ACA covariates at once